eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Check Style Rule: Good Logging Practice
+ *  I. Non-static Logger instance
+ *  II. Logger variable name should be 'logger'
+ *  III. Guard with conditional block check whenever logger's logging method called.
+ */
+public class GoodLoggingCheck extends AbstractCheck {
+    private static final String LOGGER_CLASS = "Logger";
+    private static final String LOGGER_NAME = "logger";
+
+    private static final String LOGGER_NAME_ERR = "Incorrect name for Logger: use 'logger' instead of '%s' as Logger name";
+    private static final String LOGGER_LEVEL_ERR = "Please guard %s method call with %s method";
+    private static final String STATIC_LOGGER_ERR = "Logger should not be static: remove static modifier";
+    private static final int[] TOKENS = new int[] {
+        TokenTypes.METHOD_CALL,
+        TokenTypes.VARIABLE_DEF
+    };
+
+    private static final Map<String, String> loggerMethodMap = new HashMap<>();
+
+    // Logger level static final string variables
+    private static final String FATAL = "fatal";
+    private static final String ERROR = "error";
+    private static final String WARN = "warn";
+    private static final String INFO = "info";
+    private static final String DEBUG = "debug";
+    private static final String TRACE = "trace";
+
+    private static final String IS_FATAL_ENABLED = "isFatalEnabled";
+    private static final String IS_ERROR_ENABLED = "isErrorEnabled";
+    private static final String IS_WARN_ENABLED = "isWarnEnabled";
+    private static final String IS_INFO_ENABLED = "isInfoEnabled";
+    private static final String IS_DEBUG_ENABLED = "isDebugEnabled";
+    private static final String IS_TRACE_ENABLED = "isTraceEnabled";
+
+    private static final Set<String> loggerSet = new HashSet<>();
+
+    @Override
+    public void init() {
+        initLogLevelMap();
```

Comment:
The map is static. I don't think you need to initialise it every time because only one instance of it exists.

Reply:
make sense!